client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -83,98 +144,72 @@ export class SiteVerticalsSuggestionSearch extends Component {
-		// The suggestion list should only be updated when a user is not navigating the list through keying.
-		// Note: it's intentional to use object reference comparison here.
-		// Since `verticals` props is connected from a redux state here, if the two references are identical,
-		// we can safely say that the two content are identical, thanks to the immutability invariant of redux.
-		if ( prevProps.verticals !== this.props.verticals && ! this.state.isNavigating ) {
-			// It's safe here to call setState() because we prevent the indefinite loop by the wrapping condition.
-			// See the official doc here: https://reactjs.org/docs/react-component.html#componentdidupdate
-			// eslint-disable-next-line react/no-did-update-set-state
-			this.setState( {
-				candidateVerticals: this.props.verticals,
-			} );
+		// let's update the value immediately
+		// when the user has either selected or cleared the value
+		// so we can load the preview more quickly
+		if ( isSuggestionSelected || ! value ) {
+			this.updateVerticalData( this.searchForVerticalMatches( value ), value );
 		}
-	}
 
-	getSuggestions = () => this.state.candidateVerticals.map( vertical => vertical.verticalName );
-
-	sortSearchResults = ( suggestionsArray, queryString ) => {
-		let queryMatch;
-
-		// first do the search, omit and cache exact matches
-		queryString = queryString.trim().toLocaleLowerCase();
-		const lazyResults = suggestionsArray.filter( val => {
-			if ( val.toLocaleLowerCase() === queryString ) {
-				queryMatch = val;
-				return false;
-			}
-			return val.toLocaleLowerCase().includes( queryString );
-		} );
-
-		// second find the words that start with the search
-		const startsWithResults = lazyResults.filter( val =>
-			startsWith( val.toLocaleLowerCase(), queryString )
-		);
-
-		// merge, dedupe, bye
-		return uniq(
-			startsWithResults.concat( lazyResults.concat( queryMatch ? [ queryMatch ] : [] ) )
+		this.setState(
+			{
+				isSuggestionSelected,
+				searchValue: value,
+			},
+			() => this.updateVerticalDataDebounced( this.searchForVerticalMatches( value ), value )
 		);
 	};
 
+	/***
+	 * Returns an array of vertical values - suggestions - that is consumable by `<SuggestionSearch />`
+	 *
+	 * @returns {Array} The array of vertical values.
+	 */
+	getSuggestions = () => this.state.candidateVerticals.map( vertical => vertical.verticalName );
+
 	render() {
-		const {
-			translate,
-			placeholder,
-			autoFocus,
-			shouldShowPopularTopics,
-			isVerticalSearchPending,
-		} = this.props;
-		const showPopularTopics = shouldShowPopularTopics( this.props.searchValue );
+		const { translate, placeholder, autoFocus, isVerticalSearchPending } = this.props;
 
 		return (
 			<>
-				<QueryVerticals searchTerm={ this.props.searchValue.trim() } debounceTime={ 300 } />
+				<QueryVerticals
+					searchTerm={ this.state.searchValue.trim() }
+					debounceTime={
+						! this.state.isSuggestionSelected || size( this.state.searchValue ) > 4 ? 300 : 0
```

Comment:
I personally don't quite understand the purpose behind this condition. Why do we want to alter the debouncing time based on this? How does it reflect on the UX?

Reply:
Me either. I think it was an oversight, that is, I left it in there after test. 

Thanks for catching this (again) :)